src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts

```diff
@@ -672,14 +672,14 @@ class ExtensionsContributions extends Disposable implements IWorkbenchContributi
 				}
 			],
 			icon: installWorkspaceRecommendedIcon,
-			run: () => {
-				return Promise.all(this.extensionsWorkbenchService.outdated.map(async extension => {
-					try {
-						await this.extensionsWorkbenchService.install(extension, extension.local?.preRelease ? { installPreReleaseVersion: true } : undefined);
-					} catch (err) {
-						runAction(this.instantiationService.createInstance(PromptExtensionInstallFailureAction, extension, extension.latestVersion, InstallOperation.Update, err));
+			run: async () => {
+				const results = await this.extensionsWorkbenchService.updateAll();
+				results.forEach((result) => {
+					if (result.error && result.context?.extension) {
+						const extension: IExtension = result.context.extension;
```

Comment:
I do not think this will work as the this data is sent across processes and when it is recevied back on window this might not be serialised correctly. I would recommend to collect the outdated extensions before running the updateAll and use that.

Reply:
I have pushed a change which I hope does what you mean.